sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/CommentedOutCodeCheck.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * SonarQube XML Plugin
+ * Copyright (C) 2010-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.xml.checks;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.sonar.check.Rule;
+import org.sonarsource.analyzer.commons.xml.SafetyFactory;
+import org.sonarsource.analyzer.commons.xml.XmlFile;
+import org.sonarsource.analyzer.commons.xml.XmlTextRange;
+import org.sonarsource.analyzer.commons.xml.checks.SonarXmlCheck;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+@Rule(key = CommentedOutCodeCheck.RULE_KEY)
+public class CommentedOutCodeCheck extends SonarXmlCheck {
+
+  public static final String RULE_KEY = "S125";
+
+  private final XPathExpression commentsExpression = getXPathCommentsExpression();
+
+  private final Set<Node> reportedNodes = new HashSet<>();
+
+  @Override
+  protected void scanFile(XmlFile file) {
+    Charset charset = file.getInputFile().charset();
+
+    for (Node comment : getComments(file)) {
+      if (reportedNodes.contains(comment)) {
+        // already reported in previous issue
+        continue;
+      }
+      List<Node> siblingComments = getNextCommentSiblings(comment);
+      if (isParseableXml(siblingComments, charset)) {
```

Comment:
One thing is bothering me: imagine you have a list of sibling comments.
The first one is random text -> parsing fails
So you recreate the whole list from scratch without the first comment.
Maybe isParseableXml could return a list which contains the parseable comments?